### PR TITLE
boards/serpente: add SAUL PWM integration

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

As the title says.

### Testing procedure

`make BOARD=serpente -C examples/saul flash term`

### Issues/PRs references

None

### CODE REVIEW

**Comment by chrysn** [comment]
On a superficial 30-seconds look, it looks good. If someone who has the board can test it (@madokapeng?) and comment on the results, I'll give it another look and can probably get it merged it quickly.

**Comment by madokapeng** [comment]
> On a superficial 30-seconds look, it looks good. If someone who has the board can test it (@madokapeng?) and comment on the results, I'll give it another look and can probably get it merged it quickly.

Have tested the change on serpente board.
                   R   G   B
saul write 0 255 0 0  // Red ON with full PWM
saul write 0 0 255 0  // Green ON with full PWM
saul write 0 0 0 255  // Blue ON with full PWM

Works as expected. 

out of curiosity, why does channel map to 3, 2, 4
 https://github.com/RIOT-OS/RIOT/blob/dbba60d43381017e891eef609a05ab5aa128470e/boards/serpente/include/pwm_params.h#L34-L36

**Comment by maribu** [comment]
> out of curiosity, why does channel map to 3, 2, 4

That numbers match the index in the PWM config. (Note: the first entry has index 0).

File: boards/serpente/include/pwm_params.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#include "saul/periph.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+static const saul_pwm_rgb_params_t saul_pwm_rgb_params[] =
+{
+    {
+        .name  = "RGB LED",
```
Comment (chrysn) [inline boards/serpente/include/pwm_params.h:32]: I'd rather pick "LED" as designation; that's the commonality between the LED_R, LED_G and LED_B on the pinout. (The alternative is "D3" from the schematic). It being an RGB is indicated in the SAUL type.

**Review by chrysn — APPROVED** [review]
Relying on @madokapeng's test review, and Murdock for building it, I can approve this on the remaining criteria. I recommend the indicated name change (squash right away if you do it), but it's not critical.

File: boards/serpente/include/pwm_params.h (hunk 1 above)
Comment (maribu) [reply boards/serpente/include/pwm_params.h:32]: I'm currntly afk (writing from phone). But I agree and will apply tonight.

Comment (maribu) [reply boards/serpente/include/pwm_params.h:32]: Btw: the allow edits by maintainers checkbox should be checked :-) So if you feel the itch to try that out, here is your chance ;-)

**Comment by maribu** [comment]
Thanks!

**Pull request merged**